parser/Parser.cc

```diff
@@ -24,14 +24,31 @@ class ErrorToError {
         return min((uint32_t)(pos), maxOff);
     }
 
-    static void maybeAddAutocorrect(core::ErrorBuilder &e, core::Loc loc, ruby_parser::dclass errorClass) {
+    static void maybeAddAutocorrect(core::GlobalState &gs, core::ErrorBuilder &e, core::Loc loc,
+                                    ruby_parser::dclass errorClass) {
         switch (errorClass) {
             case ruby_parser::dclass::IfInsteadOfItForTest:
                 e.replaceWith("Replace with `it`", loc, "it");
                 break;
             case ruby_parser::dclass::MissingCommaBetweenKwargs:
                 e.replaceWith("Insert a comma", loc, ", ");
                 break;
+            case ruby_parser::dclass::CurlyBracesAroundBlockPass: {
+                auto source = loc.source(gs);
+                if (source.has_value()) {
+                    auto replacement = string(loc.source(gs).value());
+                    size_t lCurlyPos = replacement.find("{");
+                    if (lCurlyPos != string::npos) {
+                        replacement = replacement.erase(lCurlyPos, 1);
+                        size_t rCurlyPos = replacement.rfind("}");
+                        if (rCurlyPos != string::npos) {
+                            replacement = replacement.erase(rCurlyPos, 1);
+                            e.replaceWith("Remove the curly braces", loc, replacement);
```

Comment:
don't we want to replace the curly braces with parentheses?

Reply:
i'm not sure that it's obvious from the examples you have in your tests that we should have an autocorrect for this. it seems like the autocorrect is only sensible for the first case (`{ &:foo }`). suggesting "remove the curly braces" doesn't actually fix the other problems, which makes for a confusing user experience

Reply:
Ooh parens are probably better, yeah!

The examples in the tests are kind of contrived (particularly the `&(1+1)` thing which is just going to go from a parse error to a type error), but I can imagine more complex cases that actually will end up type checking, like:

```
x.map {&(some_condition ? :to_s
                        : :to_i)}
```

I'm not sure how common that actually is, but autocorrecting the curlies to parens would give us something that works.